languages/python/django-oso/tests/test_post_relationship.py

```diff
@@ -541,23 +545,19 @@ def test_in_with_constraints_but_no_matching_objects(tag_nested_many_many_fixtur
         """
     )
     user = User.objects.get(username="user")
-    authorize_filter = authorize_model(None, Post, actor=user, action="read")
-    assert (
-        str(authorize_filter)
-        == "(AND: (NOT (AND: ('pk__in', []))), ('tags__name', 'bloop'))"
-    )
-    posts = Post.objects.filter(authorize_filter)
-    assert (
-        str(posts.query)
-        == 'SELECT "test_app2_post"."id", "test_app2_post"."contents", "test_app2_post"."access_level",'
-        + ' "test_app2_post"."created_by_id", "test_app2_post"."needs_moderation"'
-        + ' FROM "test_app2_post"'
-        + ' INNER JOIN "test_app2_post_tags"'
-        + ' ON ("test_app2_post"."id" = "test_app2_post_tags"."post_id")'
-        + ' INNER JOIN "test_app2_tag"'
-        + ' ON ("test_app2_post_tags"."tag_id" = "test_app2_tag"."id")'
-        + ' WHERE "test_app2_tag"."name" = bloop'
-    )
+    posts = Post.objects.authorize(None, actor=user, action="read")
+    expected = """
+        SELECT "test_app2_post"."id", "test_app2_post"."contents", "test_app2_post"."access_level", "test_app2_post"."created_by_id", "test_app2_post"."needs_moderation"
+        FROM "test_app2_post"
+        WHERE "test_app2_post"."id" IN (SELECT DISTINCT V0."id"
+                                        FROM "test_app2_post" V0
+                                        LEFT OUTER JOIN "test_app2_post_tags" V1
+                                        ON (V0."id" = V1."post_id")
+                                        WHERE EXISTS(SELECT U0."id", U0."name", U0."created_by_id", U0."is_public"
+                                                     FROM "test_app2_tag" U0
+                                                     WHERE (U0."id" = V1."tag_id" AND U0."name" = bloop)))
```

Comment:
Would need to read into it a bit more, but at first blush going from a pair of `INNER JOIN`s to an `IN` + a `LEFT OUTER JOIN` + an `EXISTS` seems like a regression.

Reply:
I sort of agree, and it would be straightforward to revert that change. The reason I liked it, is because:
-  it matches the Polar semantics way closer
- there's no need to differentiate between an in with constraints, and one without
- the intersection case will "just work" (hopefully)
- and when we do negation, you can just negate the exists and you're done.